src/bitbucket/BitbucketClient.ts

```diff
@@ -58,7 +58,18 @@ export class BitbucketClient {
 
     if (!approvalChecks.isOpen) {
       errors.push('PR is already closed!');
+    } else if (prSettings.rules) {
+      const pullRequestInfo = {
+        pullRequest,
+        buildStatuses,
+        approvals,
+        permissionLevel,
+      };
+      errors.push(
+        ...prSettings.rules.filter(({ rule }) => !rule(pullRequestInfo)).map(({ error }) => error),
```

Comment:
This is a neat way of running them but is a bit tricky to parse imo.

Could we write this a bit clearer:

```
prSettings.customRules.forEach(({ rule, errorMsg }) => {
  const passesRule = rule(pullRequestInfo);
  if (!passesRule) errors.push(errorMsg);
});
```